File: LC/Python/Medium/55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        max_pos = len(nums) - 1
        
        for i in range(len(nums) - 1, -1, -1):
            if i + nums[i] >= max_pos:
                max_pos = i
        
        return True if max_pos == 0 else False

File: LC/C++/Medium/55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int maxPos = n - 1;
        
        for (int i = n - 1; i >= 0; --i) {
            if (i + nums[i] >= maxPos) {
                maxPos = i;
            }
        }

        return maxPos == 0;
    }
};